### PR TITLE
Pptp 1178 capture business address

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
PPTP-1178 : Capture Business Address - Part I

Refactored to ensure that the "registered business address" is populated in the GRS callback via methods on Registration and OrganisationDetails rather than being done externally during display of the address confirmation page.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala [hunk 1]
```diff
@@ -161,81 +161,74 @@ class GrsController @Inject() (
...
+  ): Future[Either[ServiceError, Registration]] = {
+    val updatedRegistration = request.registration.organisationDetails.organisationType match {
       case Some(OrgType.UK_COMPANY) | Some(OrgType.OVERSEAS_COMPANY_UK_BRANCH) =>
         updateUkCompanyDetails(journeyId)
       case Some(OrgType.REGISTERED_SOCIETY) => updateRegisteredSocietyDetails(journeyId)
       case Some(OrgType.SOLE_TRADER)        => updateSoleTraderDetails(journeyId)
       case Some(OrgType.PARTNERSHIP)        => updatePartnershipDetails(journeyId)
       case _                                => throw new InternalServerException(s"Invalid organisation type")
     }
+    updatedRegistration
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala:173]: Maybe drop the intermediate `updatedRegistration` val? It flows better if the flatMap occurs on the match result.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala [hunk 2]
```diff
@@ -113,6 +113,9 @@ case class Registration(
 
   val isFirstGroupMember = groupDetail.exists(_.members.isEmpty)
 
+  def populateBusinessRegisteredAddress(): Registration =
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala:116]: Does this belong here? It feels like it should be in `RegistrationBuilder`? There seems to be some blurring of responsibilities between `RegistrationBuilder` and the `Registration` object.

**Review by paulblakehmrc — APPROVED** [review]
Good idea to refactor this. More natural place for this code to live. A couple of minor optional points.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala (hunk 2 above)
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala:116]: scratch that - builder is test-only

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala (hunk 1 above)
Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/organisation/GrsController.scala]: Done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala [hunk 3]
```diff
@@ -107,26 +107,6 @@ case class OrganisationDetails(
...
-        Some(
-          Address(addressLine1 = "3 Scala Street",
-                  addressLine2 = Some("Soho"),
-                  townOrCity = "London",
-                  postCode = partnershipDetails.flatMap(
-                    pd => pd.partnershipBusinessDetails.map(pbd => pbd.postcode)
-                  )
-          )
-        )
       case _ => None
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:110]: Finally, Scala Street is history, in FE anyway.

File: tampermonkey/PPT_AutoComplete.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // ==UserScript==
 // @name         PPT Registration AutoComplete
 // @namespace    http://tampermonkey.net/
-// @version      15.2
+// @version      15.3
```
Comment (paulblakehmrc) [inline tampermonkey/PPT_AutoComplete.js:4]: is this the only change to the script?

**Review by paulblakehmrc — APPROVED** [review]
LGTM. Just one query on the TamoerMonkey script.

File: tampermonkey/PPT_AutoComplete.js (hunk 4 above)
Comment (RyanKellyHMRC) [reply tampermonkey/PPT_AutoComplete.js:4]: My mind must have wandered off. I thought I needed to make some changes but I decided that I did not. I'll revert.

**Pull request merged**